### PR TITLE
Added availableItemsNotifier

### PR DESCRIPTION
**Pull request by zarutian**
Groundwork for [dapp-card-store issue #19](https://github.com/Agoric/dapp-card-store/issues/19).

By the way, I found it a bit odd that there is no way, yet, to add to the sellItems Items inventory. Is this supposed to be this way?

### CODE REVIEW

**Comment by zarutian** [comment]
Paging @katelynsills or @michaelfig for review.

File: packages/zoe/src/contracts/sellItems.js [hunk 1]
```diff
@@ -42,10 +43,19 @@ const start = zcf => {
...
+  /** @type {undefined | ZCFSeat} */
   let sellerSeat;
 
+  const { notifier: availableItemsNotifier,
+          updater: availableItemsUpdater } = makeNotifierKit();
+
   const sell = seat => {
     sellerSeat = seat;
+    observeIteration(sellerSeat.getNotifier(), harden({
+      updateState: sellerSeatAllocation => availableItemsUpdater.updateState(sellerSeatAllocation['Items']),
```
Comment (katelynsills) [inline packages/zoe/src/contracts/sellItems.js]: Can we use dot notation instead? Same below for `finish`

```suggestion
      updateState: sellerSeatAllocation => availableItemsUpdater.updateState(sellerSeatAllocation.Items),
```

**Review by katelynsills — COMMENTED** [review]
Thanks so much @zarutian. One suggestion about dot notation. Also, can you add a test for this? There should be a test file called something like `test-sellItems.js`. There should be a test for `getAvailableItems` in there. Can you add a test for the notifier nearby? Thanks!

**Comment by zarutian** [comment]
I could only find $git-root$/packages/zoe/test/unitTests/contracts/test-sellTickets.js as possible candidate. Is that the one you mean @katelynsills ?

**Comment by katelynsills** [comment]
> I could only find $git-root$/packages/zoe/test/unitTests/contracts/test-sellTickets.js as possible candidate. Is that the one you mean @katelynsills ?

Yup, that's right. I forgot we had no test of `sellItems` directly. Can you add a test in `test-sellTickets.js`?

**Comment by zarutian** [comment]
A bit stumped  with how to test this.

The only scenario that comes to mind is:
1. Alice instanciates the sellItems contract.
2. she gets the `availableItemsNotifier` via `publicFacet.getAvailableItemsNotifer()`
3. she mints some NFT say `goldenTurds`
3. using the `creatorInvotation` she got from the contract instanciation, she adds those aforesaid goldenTurds and wants x moola in return.
4. then what?

**Comment by katelynsills** [comment]
That scenario sounds...er.. good. 😅 

Another scenario is that Bob gets a notifier for the available items and then Carol buys a ticket and Bob's notifier provides a notification that the available items have changed. 

Actually, lines 207-250 in test-sellTickets.js can be used, if you just want to get a notifier and check the available items before alice buys a ticket, and then check them again afterward.

**Comment by zarutian** [comment]
@katelynsills ready for first review

**Comment by katelynsills** [comment]
Hey @zarutian, this looks amazing, and I think you found a bug in Zoe. The notifier should start with the initial allocation.

I have a few edits in a commit here: https://github.com/Agoric/agoric-sdk/tree/sell-items-notifier 

Can you do me a favor and select: `Allow edits from maintainers` (more info here: https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) 

I believe this will let me add my commits to your PR, and then we can merge this. Thanks so much!

**Comment by zarutian** [comment]
Huh, I thought I had allowed edits by maintainers. Well I toggled it and it should work now?

**Review by katelynsills — APPROVED** [review]
Looks good to me! Thanks so much @zarutian!

**Comment by zarutian** [comment]
Ya have something against turds? Specially golden ones that are often a token of good fortune, eh? ;-þ

I got the idea from an Japanese Game Show anime and it turns out to be real.

Though I understand perfectly why you changed it to birds though.

**Pull request merged**